### DISCUSSION TITLE
Code Browser function failing to display Gitlab project

### DISCUSSION THREAD

**Discussion by factor3** [TabbyML/tabby#2400] [General]
In my development lab, I have a self- hosted Gitlab server with several projects. In order to use the functions of the Browser, I created the  necessary project token, successfully connected the Tabby to the project, and clicked on the "Browser" link to initiate the code browser. Selecting my Gitlab project, I get the screen shown below:

![tabbyprob](https://github.com/TabbyML/tabby/assets/138332567/1ac6684a-1ba7-497f-b80f-b99257abc7ee)

The display shows no files, no subfolders, just a single folder. Clicking on the folder just brings me to the root.

It looks like Tabby is trying to scan my repository, and getting stuck doing so. That is what it looks like, I do not know for sure.

Ultimately, the code browser is not working. Can someone give some idea of what is wrong and how I can fix it?

**Comment by wsxiaoys** [comment]
Hi, could you check the `Jobs` tab in admin UI to see if the job finished for importing your repos?

**Reply by factor3** [reply]
Grreetings:

I checked the Jobs tab. I saw a large number of "finished" jobs that apparently took zero seconds to complete. Checking each job, I am seeing that each job was for "refreshing" my repository. I checked the "stderr" tab for each one. They all said "no data".

Tabby is clearly attempting to update its info on my repo, and it is failing to read the repo.

The instructions for setting up the browser said to set the token permissions to "api" and "read repository". I did set those permissions when I created my project token. Is there another permission or set of permissions that I should set?

**Comment by wsxiaoys** [comment]
Filing https://github.com/TabbyML/tabby/issues/2401

**Comment by factor3** [comment]
I have downloaded (what I believe is) the latest TabbyML container.  I am still getting the same problem with the failure to display the Gitlab project. I am also getting the same "no data" in the Jobs tab as well.

One thing I am noticing, though: when I set up my URL for the connection to my Gitlab repository, I set it up as an HTTPS. But when I add a repository, it always appears as HTTP.

My access to my Gitlab is through HTTPS. Any attempt to access it through HTTP attempts to force the requestor to retry with HTTPS. Could this create a problem for the UI? Could the UI be attempting to access the repository using HTTP and failing to gather data due to the inability to access it using HTTPS?

If it does, then there needs to be a way to change the HTTP to using HTTPS. There does not appear to be any way to do this right now.

**Reply by wsxiaoys** [reply]
Please attach the version information. You can find it in 'System' tab in admin UI.

**Reply by factor3** [reply]
The version is: v0.12.0.

That is strange. This was released 2 weeks ago (according to the page https://github.com/TabbyML/tabby/releases/tag/v0.12.0). It is also showing as being the latest version.

Has the latest version, due for the 21st, not been released yet?

**Reply by wsxiaoys** [reply]
0.13 hasn't been released yet - if you wanna give RC version a try, please use docker image tag 0.13.0-rc.1

**Reply by factor3** [reply]
Thank you for the tag.

I have upgraded to the  0.13.0-rc.1. I have found some interesting problems:

1. I am noticing that when I go to Settings->Context Providers->Gitlab Self- Hosted, I see the newly created provider. Unfortunately, when  I click on the "View" button, I see that my instance URL and my Personal Access Token are no longer there. The Display Name is there, though.
2.  I put in the instance URL and the Personal Access Token again, I click on the Update button, then I wait for a few minutes in order for the  synchronizing of the connection to continue. I then try to add a Repository from the provider. Unfortunately, the Add New Repository  dialog box displays "No repository found" -- despite the fact that there is a project that I have been attempting to use the Browser function with several times before.
3. Of course, if I look again at my instance URL and the Personal Access Token, either by refreshing the page or displaying it through  the View button, The instance URL and Personal Access Token disappear again.

It looks like the (Beta) Playground system is working pretty well.

**Reply by wsxiaoys** [reply]
For issue 1 - can you share a tab recording (e.g using https://jam.dev)? Thanks!
